lang/src/org/partiql/lang/eval/physical/window/Lag.kt

```diff
@@ -0,0 +1,40 @@
+package org.partiql.lang.eval.physical.window
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.numberValue
+import org.partiql.lang.eval.physical.EvaluatorState
+import org.partiql.lang.eval.physical.operators.ValueExpression
+
+// TODO: Remove from experimental once https://github.com/partiql/partiql-docs/issues/31 is resolved and a RFC is approved
+@ExperimentalWindowFunc
+internal class Lag : NavigationWindowFunction() {
+    override val name = "lag"
+
+    override fun processRow(state: EvaluatorState, arguments: List<ValueExpression>, currentPos: Int): ExprValue {
+        val (target, offset, default) = when (arguments.size) {
+            1 -> listOf(arguments[0], null, null)
+            2 -> listOf(arguments[0], arguments[1], null)
+            3 -> listOf(arguments[0], arguments[1], arguments[2])
+            else -> error("Wrong number of Parameter for Lag Function")
+        }
+
+        val offsetValue = offset?.let {
+            val numberValue = it.invoke(state).numberValue().toLong()
+            if (numberValue >= 0) {
+                numberValue
+            } else {
+                error("offset need to be non-negative integer")
+            }
+        } ?: 1L // default offset is one
```

Comment:
Should this be a constant?

Reply:
In the usual case it will be, but there is nothing in the Sql Spec that prohibit one from using a variable here. 
See this (rather weird example) https://www.db-fiddle.com/f/8KQkkdZot6j7gdWanqB5g8/0

Reply:
Right but `?: DEFAULT_VALUE`